// Copyright (c) farm-ng, inc.
//
// Licensed under the Amiga Development Kit License (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://github.com/farm-ng/amiga-dev-kit/blob/main/LICENSE
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package farm_ng.people_detection.proto;

service PeopleDetectionService {
  rpc detectPeople(DetectPeopleRequest) returns (DetectPeopleReply) {}
}

message Detection {
  float x = 1;
  float y = 2;
  float width = 3;
  float height = 4;
  float confidence = 5;
}

message ImageSize {
  int32 width = 1;
  int32 height = 2;
}

message DetectPeopleConfig {
  float confidence_threshold = 1;
}

message DetectPeopleRequest {
  DetectPeopleConfig config = 1;
  ImageSize image_size = 2;
  bytes image_data = 3;
}

message DetectPeopleReply {
  repeated Detection detections = 1;
}
